lib/journaled/connection.rb

```diff
@@ -0,0 +1,48 @@
+module Journaled
+  module Connection
+    class << self
+      def available?
+        Journaled.transactional_batching_enabled && transaction_joinable?
```

Comment:
This object has two purposes: to tell the journaled writer whether there is a transactional context available, and (if so) provide it with the blessed way of staging events onto that transactional context.